### ISSUE TITLE
Error in production version minification in next 13, error not in next 12 or dev version of next 13

### ISSUE THREAD

**Issue by cmdcolin** [vercel/next.js#43208]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```
yarn next info
yarn run v1.22.19
$ /home/cdiesh/out/node_modules/.bin/next info

    Operating System:
      Platform: linux
      Arch: x64
      Version: #25-Ubuntu SMP PREEMPT_DYNAMIC Mon Nov 14 14:06:02 UTC 2022
    Binaries:
      Node: 18.12.1
      npm: 8.19.2
      Yarn: 1.22.19
      pnpm: N/A
    Relevant packages:
      next: 13.0.5-canary.3
      eslint-config-next: 13.0.4
      react: 18.2.0
      react-dom: 18.2.0

Done in 0.97s.

```

### What browser are you using? (if relevant)

Chrome 107.0.5304.110, also Firefox 107.0

### How are you deploying your application? (if relevant)

next start

### Describe the Bug

I have an app where in dev mode (next dev) the app works, but next build+next start appears to introduce a minification bug. It works with next 12.x but appeared to cause issue with next 13. I don't have a super minimal example (yet) but can try to find one if interested (and similarly, move to another repo like swc if it makes sense?)

### Expected Behavior

Minification should produce a valid program

### Link to reproduction - Issues with a link to complete (but minimal) reproduction code will be addressed faster

https://github.com/cmdcolin/nextjs-13-minify-error

### To Reproduce

I reduced the example somewhat to the usage of (our) library https://github.com/GMOD/cram-js/ and made a smaller reproducible example here https://github.com/cmdcolin/nextjs-13-minify-error

The TLDR of how this example was made was just a simple create-next-app with next 13, add @gmod/cram library, add basic usage of this library to pages/index.tsx, and run in production produces the syntax error from bad minification. I did `yarn add next@canary` and had the same bug

The app works with next 12 though

I could possibly make the reproducible case more minimal if interested

**Comment by cmdcolin** [comment]
fixed title :)

**Comment by ijjk** [comment]
Hi, can you confirm this issue is still present in `v13.0.5-canary.4` of Next.js?

**Comment by cmdcolin** [comment]
I believe it is still happening with canary 4. I pushed this latest canary to my repro

**Comment by balazsorban44** [comment]
I could confirm on `canary` too, we will have a look, thanks!

**Comment by cmdcolin** [comment]
I reduced the example a little more, so now it is just using @gmod/binary-parser (which is our fork of https://github.com/keichi/binary-parser). Interestingly, I found the normal https://github.com/keichi/binary-parser does not have the issue. both @gmod/binary-parser and binary-parser packages uses `eval`, and so the issue may be related to minifying the eval'd code presumably

**Comment by kdy1** [comment]
I'm working on this at https://github.com/swc-project/swc/pull/6507 , but I'm not sure how I should create a test case for it.

**Comment by kdy1** [comment]
This will be fixed by https://github.com/swc-project/swc/pull/6521

**Comment by kdy1** [comment]
Closing as this is fixed by https://github.com/vercel/next.js/pull/43652
